bayes_opt/target_space.py

```diff
@@ -265,45 +265,55 @@ def random_sample(self):
             data.T[col] = self.random_state.uniform(lower, upper, size=1)
         return data.ravel()
 
+    def y_max(self):
+        """Get maximum target value found.
+        
+        If there is a constraint present, the maximum value that fulfills the
+        constraint is returned."""
+        if len(self.target) == 0:
+            return None
+
+        if self._constraint is None:
+            return self.target.max()
+
+        allowed = self._constraint.allowed(self._constraint_values)
+        if allowed.any():
+            return self.target[allowed].max()
+
+        return None
+
     def max(self):
         """Get maximum target value found and corresponding parameters.
         
         If there is a constraint present, the maximum value that fulfills the
         constraint is returned."""
-        if self._constraint is None:
-            try:
-                res = {
-                    'target': self.target.max(),
-                    'params': dict(
-                        zip(self.keys, self.params[self.target.argmax()])
-                    )
-                }
-            except ValueError:
-                res = {}
-            return res
-        else:
-            allowed = self._constraint.allowed(self._constraint_values)
-            if allowed.any():
-                # Getting of all points that fulfill the constraints, find the
-                # one with the maximum value for the target function.
-                sorted = np.argsort(self.target)
-                idx = sorted[allowed[sorted]][-1]
-                # there must be a better way to do this, right?
-                res = {
-                    'target': self.target[idx],
-                    'params': dict(
-                        zip(self.keys, self.params[idx])
-                    ),
-                    'constraint': self._constraint_values[idx]
-                }
-            else:
-                res = {
-                    'target': None,
-                    'params': None,
-                    'constraint': None
-                }
+        y_max = self.y_max()
+
+        if y_max is None:
+            res = {
+                'target': None,
+                'params': None
+            }
+
+            if self._constraint is not None:
+                res['constraint'] = None
```

Comment:
I guess this is a breaking change, if you wan't me to keep the old behaviour I can add another commit.

I did it like this because in the old behaviour we had a `res = {}` for unconstrained case and `res = {'target': None, ]params': None, 'constraint': None}` for the constrained case and I wanted to unify the behaviour.

Reply:
I agree - both that None is better and also that either is ok :-). This still seems to be returning a dict of None at the moment?

Reply:
Let's go with `None` then. That should be the last thing and then we can merge this!

Reply:
Done